src/main/java/s/volodymyr/onlinebookstore/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,100 @@
+package s.volodymyr.onlinebookstore.service.impl;
+
+import java.util.HashSet;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import s.volodymyr.onlinebookstore.dto.cartitem.CreateCartItemRequestDto;
+import s.volodymyr.onlinebookstore.dto.shoppingcart.ChangeShoppingCartQuantityDto;
+import s.volodymyr.onlinebookstore.dto.shoppingcart.ShoppingCartDto;
+import s.volodymyr.onlinebookstore.exception.EntityNotFoundException;
+import s.volodymyr.onlinebookstore.mapper.ShoppingCartMapper;
+import s.volodymyr.onlinebookstore.model.Book;
+import s.volodymyr.onlinebookstore.model.CartItem;
+import s.volodymyr.onlinebookstore.model.ShoppingCart;
+import s.volodymyr.onlinebookstore.model.User;
+import s.volodymyr.onlinebookstore.repository.book.BookRepository;
+import s.volodymyr.onlinebookstore.repository.cartitem.CartItemRepository;
+import s.volodymyr.onlinebookstore.repository.shoppingcart.ShoppingCartRepository;
+import s.volodymyr.onlinebookstore.repository.user.UserRepository;
+import s.volodymyr.onlinebookstore.service.ShoppingCartService;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final UserRepository userRepository;
+    private final BookRepository bookRepository;
+    private final CartItemRepository cartItemRepository;
+
+    @Transactional
+    @Override
+    public ShoppingCartDto findById(Long id) {
+        return shoppingCartMapper.toDto(shoppingCartRepository.findById(id).orElseThrow(
+                () -> new RuntimeException("Cannot find shopping cart")
+        ));
+    }
+
+    @Transactional
+    @Override
+    public ShoppingCartDto addToCart(Long id, CreateCartItemRequestDto requestDto) {
```

Comment:
pls specify that this id belongs to user to avoid confusion
```suggestion
    public ShoppingCartDto addToCart(Long userId, CreateCartItemRequestDto requestDto) {
```

makes sense to adjust in all places